executor/src/witgen/generator.rs

```diff
@@ -41,6 +41,22 @@ enum EvaluationRow {
     Next,
 }
 
+#[derive(PartialEq, Eq, Clone, Copy)]
+enum SolvingStrategy {
+    /// Only solve expressions that are affine in a single variable
+    /// (and use bit constraints).
+    SingleVariableAffine,
+    /// Assume that all unknown values are zero and check that this does not generate
+    /// a conflict (but do not store the values as fixed zero to avoid relying on nondeterminism).
```

Comment:
Regarding this comment, any kind of more complex solver we add would have to rely on nondeterminism, no?

Like, a solver that only looks at one div equation `divisor * QUOT + REM = dividend` would have a whole 1-D curve of solutions.

Reply:
yes of course. but if we are lucky, there might be bit constraints. I didn't analyze the problem fully yet, it could be that bit constraints do not always rule out other solutions to the individual div equation.

I mean in the end of course it should not need to rely on nondeterminism, but it might be difficult to check.